shuffle/sample-app/src/main.rs

```diff
@@ -108,3 +174,65 @@ fn send(client: &BlockingClient, tx: diem_types::transaction::SignedTransaction)
     );
     Ok(())
 }
+
+/// Compile Move program
+pub fn compile_program(file_path: &str, dependency_paths: &[&str]) -> Result<Vec<u8>> {
+    let tmp_output_dir = TempPath::new();
+    tmp_output_dir
+        .create_as_dir()
+        .expect("error creating temporary output directory");
+    let tmp_output_path = tmp_output_dir.as_ref().display().to_string();
+
+    let mut command = Command::new("cargo");
+    command
+        .args(&["run", "-p", "move-lang", "--bin", "move-build", "--"])
```

Comment:
This is a quick prototype, but why invoke the `cargo run` cli when you can just invoke the underlying methods directly?

Reply:
After further thought, can we please invoke this directly via rust, in memory, rather than spinning up a process to invoke a cargo cmd?